### PR TITLE
Stry 28 contractor profile backend

### LINKED ISSUE #72: Task: Contractor Profile Schema (Backend)
This task covers defining the CO profile schema in the backend and driving the subsequent logic.

### PR DESCRIPTION
**Pull request by and-le**
Defines the schema for contractors and includes a GET and PATCH endpoint for their profile information.

I also included testing changes. To summarize, I updated our GitHub workflows to run two sets of tests separately, one for the frontend and one for the backend.

Tested using the tests in the backend `tests` directory.

Apologies for the large number of files changed, but I needed to modify/delete/move them to get the testing changes to work.

The listings tests are skipped because they fail whenever they are run with other tests. The cause may be Jest itself, as race conditions appear to be causing the failures. I may have to migrate the testing to Mocha.

Closes #72

### CODE REVIEW

**Comment by Rebooting-Me** [comment]
Looks good to me for merging.

**Review by Rebooting-Me — APPROVED** [review]

File: backend/models/contractorModel.js [hunk 1]
```diff
@@ -30,14 +29,14 @@ const contractorSchema = new Schema({
     required: false
   },
   // The services that this contractor provides
-  // TODO: figure out how services will be checked on backend
   services: {
```
Comment (Rebooting-Me) [inline backend/models/contractorModel.js:32]: Do you want to keep this format or `service: {
    type: [String],
    enum: getServices(),
    required: false
  },` I will update my code accordingly then.

**Review by Rebooting-Me — CHANGES_REQUESTED** [review]

File: backend/models/contractorModel.js (hunk 1 above)
Comment (and-le) [reply backend/models/contractorModel.js:32]: I updated the code, so I'm not sure what line you are referring to. However, if you're referring to the definition of services, it should be the following:


services: {
    type: [
      {
        type: String,
        enum: getServices()
      }
    ],
    required: false,
  }

The previous definition would not have used the enum validator the way we intended.

Comment (Rebooting-Me) [reply backend/models/contractorModel.js:32]: Ok I will update my CO service def to this.

**Comment by Rebooting-Me** [comment]
You can go ahead and merge this.

**Review by Rebooting-Me — APPROVED** [review]

**Pull request merged**